python/llm/src/ipex_llm/transformers/pipeline_parallel.py

```diff
@@ -885,7 +919,11 @@ async def process_step(self, tokenizer, result_dict):
 
         output, cur_batch = self.model_step(cur_input, cur_batch)
 
-        self.send_buff = output
+        if self.send_buff is not None:
+            torch.xpu.synchronize(self.device)
+            self.send_buff.wait()
+        if output is not None:
+            self.send_buff = dist.isend(output, dst=self.next_rank)
```

Comment:
https://pytorch.org/docs/stable/distributed.html#torch.distributed.isend

Use async `isend` to replace sync `send`.